src/mlpack/core/data/load_deprecated.hpp

```diff
@@ -0,0 +1,462 @@
+/**
+ * @file core/data/load_deprecated.hpp
+ * @author Omar Shrit
+ * @author Ryan Curtin
+ *
+ * Shim to the old deprecated load() functions defined in load.hpp.
+ *
+ * This file should be removed when releasing mlpack 5.0.0
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_LOAD_DEPCRECATED_HPP
+#define MLPACK_CORE_DATA_LOAD_DEPCRECATED_HPP
+
+// In case it hasn't already been included.
+#include "load.hpp"
+
+#include "image_options.hpp"
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Loads a matrix from file, guessing the filetype from the extension.  This
+ * will transpose the matrix at load time (unless the transpose parameter is set
+ * to false).
+ *
+ * The supported types of files are the same as found in Armadillo:
+ *
+ *  - CSV (arma::csv_ascii), denoted by .csv, or optionally .txt
+ *  - TSV (arma::raw_ascii), denoted by .tsv, .csv, or .txt
+ *  - ASCII (arma::raw_ascii), denoted by .txt
+ *  - Armadillo ASCII (arma::arma_ascii), also denoted by .txt
+ *  - PGM (arma::pgm_binary), denoted by .pgm
+ *  - PPM (arma::ppm_binary), denoted by .ppm
+ *  - Raw binary (arma::raw_binary), denoted by .bin
+ *  - Armadillo binary (arma::arma_binary), denoted by .bin
+ *  - HDF5 (arma::hdf5_binary), denoted by .hdf, .hdf5, .h5, or .he5
+ *
+ * By default, this function will try to automatically determine the type of
+ * file to load based on its extension and by inspecting the file.  If you know
+ * the file type and want to specify it manually, override the default
+ * `inputLoadType` parameter with the correct type above (e.g.
+ * `arma::csv_ascii`.)
+ *
+ * If the detected file type is CSV (`arma::csv_ascii`), the first row will be
+ * checked for a CSV header.  If a CSV header is not detected, the first row
+ * will be treated as data; otherwise, the first row will be skipped.
+ *
+ * If the parameter 'fatal' is set to true, a std::runtime_error exception will
+ * be thrown if the matrix does not load successfully.  The parameter
+ * 'transpose' controls whether or not the matrix is transposed after loading.
+ * In most cases, because data is generally stored in a row-major format and
+ * mlpack requires column-major matrices, this should be left at its default
+ * value of 'true'.
+ *
+ * @param filename Name of file to load.
+ * @param matrix Matrix to load contents of file into.
+ * @param fatal If an error should be reported as fatal (default false).
+ * @param transpose If true, transpose the matrix after loading (default true).
+ * @param inputLoadType Used to determine the type of file to load (default arma::auto_detect).
+ * @return Boolean value indicating success or failure of load.
+ */
+template<typename eT>
```

Comment:
Maybe mark these as `[[deprecated("Will be removed in mlpack 5.0.0.  Pass a DataOptions instead.")]]`?